### ISSUE TITLE
Warn if invalid keyword arguments are passed to a function

### ISSUE THREAD

**Issue by ssssam** [mesonbuild/meson#789]
The following meson.build file triggers no errors, either with Meson 0.31.0 or with current 'master' commit a2e7ebc575a47bfb3:

```
project('testcase', 'c', foo: 'nonsense')
declare_dependency(grr: 1, bark: 2, woof: 3)
```

This is causing issues for me because I'll do something dumb like this (in a real project) without realising:

```
 widgets_dep = declare_dependency(
  link_with: librbwidgets,
  declare_dependencies: [glib, gstreamer, gtk]
)
```

and then I'll run a build and see that something, somewhere went wrong in the long chain of dependencies within this project and the Gtk compile flags are being lost ... but with no hint as to where the error is, it often takes a while debugging to figure that out.

https://github.com/mesonbuild/meson/issues/3 suggests that Meson would warn on invalid keyword arguments, but it doesn't.

**Comment by ssssam** [comment]
Seems that it does work for some functions, but not others (maybe just not for declare_dependency )

**Comment by eli-schwartz** [comment]
Fixed in #2001 -- current versions of meson annotate each interpreter function with the arguments that are considered valid:

```
The Meson build system
Version: 0.56.2
Source dir: /tmp
Build dir: /tmp/builddir
Build type: native build
../meson.build:1: WARNING: Passed invalid keyword argument "foo".
WARNING: This will become a hard error in the future.
Project name: testcase
Project version: undefined
C compiler for the host machine: ccache cc (gcc 10.2.0 "cc (GCC) 10.2.0")
C linker for the host machine: cc ld.bfd 2.35.1
Host machine cpu family: x86_64
Host machine cpu: x86_64
../meson.build:2: WARNING: Passed invalid keyword argument "grr".
WARNING: This will become a hard error in the future.
../meson.build:2: WARNING: Passed invalid keyword argument "bark".
WARNING: This will become a hard error in the future.
../meson.build:2: WARNING: Passed invalid keyword argument "woof".
WARNING: This will become a hard error in the future.
Build targets in project: 0
```
